fuse/file_test.go

```diff
@@ -0,0 +1,653 @@
+package fuse
+
+import (
+	"context"
+	"testing"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"github.com/puppetlabs/wash/datastore"
+	"github.com/puppetlabs/wash/plugin"
+	plugintest "github.com/puppetlabs/wash/plugin/test"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type fileTestSuite struct {
+	suite.Suite
+	ctx context.Context
+}
+
+func (suite *fileTestSuite) SetupTest() {
+	plugin.SetTestCache(datastore.NewMemCache())
+}
+
+func (suite *fileTestSuite) TearDownTest() {
+	plugin.UnsetTestCache()
+}
+
+func (suite *fileTestSuite) TestOpen_File() {
+	var req fuse.OpenRequest
+	var resp fuse.OpenResponse
+
+	m := plugintest.NewMockBase()
+	mr := plugintest.NewMockRead()
+	mr.Attributes().SetSize(1)
+	mw := plugintest.NewMockWrite()
+	mw.Attributes().SetSize(1)
+	mrw := plugintest.NewMockReadWrite()
+	mrw.Attributes().SetSize(1)
+
+	// expectErrs corresponds to expectations on fields in flags.
```

Comment:
This took me a bit to figure out what you mean, so I think an example would be useful here.